modules/misc/autoscreenshot.lua

```diff
@@ -0,0 +1,26 @@
+--[[
+    modules\misc\autoscreenshot.lua
+    Automatically takes a screenshot when an achievement is earned
+]]
+local addonName, Loc = ...;
+
+local AutoScreenshotFrame = CreateFrame('Frame', nil, UIParent);
+
+--[[
+    Handles the WoW API Events Registered Below
+
+    @ param Frame $self The Frame that is handling the event 
+    @ param string $event The WoW API Event that has been triggered
+    @ param arg $... The arguments of the Event
+    @ return void
+]]
+local function HandleEvents (self, event, ...)
+    if (event == 'ACHIEVEMENT_EARNED' and PrimaryDB.autoScreenshot) then
+        Screenshot();
+    end
+    -- Should we handle SCREENSHOT_SUCCEEDED / SCREENSHOT_FAILED, probably not our business.

```

Comment:
Figured handling `SCREENSHOT_SUCCEEDED` or `SCREENSHOT_FAILED` was a bit over-reaching for what this should be doing, figured we should just try and leave it at that, maybe print a message to chat afterwards?